### PR TITLE
Add needed events to recovery mode; Improve event tests

### PR DESCRIPTION
**Pull request by area**
Based on the designs for the recovery mode screens, we need these events, so I've added them.

While writing the tests it also became apparent that we were not rigorously testing our events. So I've improved `expectEvent` to check the parameters passed by the event as well, and updated all relevant tests (we probably need more, but not in this PR).

### CODE REVIEW

File: contracts/common/ContractRecovery.sol [hunk 1]
```diff
@@ -82,6 +88,7 @@ contract ContractRecovery is ContractRecoveryDataTypes, CommonStorage { // ignor
     uint numRequired = totalAuthorized / 2 + 1;
     require(recoveryApprovalCount >= numRequired, "colony-recovery-exit-insufficient-approvals");
     recoveryMode = false;
+    emit RecoveryModeExited();
```
Comment (kronosapiens) [inline contracts/common/ContractRecovery.sol]: Can we add a newline above as per convention?

File: contracts/common/ContractRecovery.sol [hunk 2]
```diff
@@ -68,7 +70,7 @@ contract ContractRecovery is ContractRecoveryDataTypes, CommonStorage { // ignor
     recoveryApprovalCount = 0;
     recoveryEditedTimestamp = block.timestamp;
 
-    emit RecoveryModeEntered();
+    emit RecoveryModeEntered(msg.sender);
```
Comment (kronosapiens) [inline contracts/common/ContractRecovery.sol:73]: Isn't this information already available on the tx?

File: helpers/test-helper.js [hunk 3]
```diff
@@ -251,9 +251,21 @@ export async function getBlockTime(blockNumber) {
   return p;
 }
 
-export async function expectEvent(tx, eventName) {
+function hexlifyAndPad(input) {
```
Comment (kronosapiens) [inline helpers/test-helper.js:254]: `hexify`?

File: test/contracts-network/colony-task-work-rating.js [hunk 4]
```diff
@@ -354,7 +354,7 @@ contract("Colony Task Work Rating", (accounts) => {
       );
     });
 
-    it("should log a TaskWorkRatingRevealed event", async () => {
+    it.only("should log a TaskWorkRatingRevealed event", async () => {
```
Comment (kronosapiens) [inline test/contracts-network/colony-task-work-rating.js:357]: 🚨 🚨 🚨

File: contracts/common/ContractRecovery.sol (hunk 2 above)
Comment (area) [reply contracts/common/ContractRecovery.sol:73]: I originally didn't have this, but ended up including it - `tx.origin` is available, but if the permission holder is a contract, it is not easily available.

File: helpers/test-helper.js (hunk 3 above)
Comment (area) [reply helpers/test-helper.js:254]: It's called `hexlify` in ethers.

File: test/contracts-network/colony-task-work-rating.js (hunk 4 above)
Comment (area) [reply test/contracts-network/colony-task-work-rating.js:357]: :sob:

**Review by kronosapiens — APPROVED** [review]

**Pull request merged**